### PR TITLE
Add opentelemetry.io docs

### PR DESCRIPTION
**Pull request by austinlparker**
Per open-telemetry/opentelemetry.io#472, we're mirroring the docs content on the website to each SIG. When a release occurs and these docs are updated, please make an issue or PR mirroring them to their appropriate location in the website repo (https://github.com/open-telemetry/opentelemetry.io/tree/main/content/en/docs/cpp).

### CODE REVIEW

File: website_docs/_index.md
Comment (lalitb) [inline website_docs/_index.md:21]: nit - we do have logging api/sdk pre-alpha release.

File: website_docs/_index.md
Comment (lalitb) [inline website_docs/_index.md:28]: API docs: https://opentelemetry-c-api.readthedocs.io/en/latest/index.html
Contrib repo : https://github.com/open-telemetry/opentelemetry-cpp-contrib

**Comment by reyang** [comment]
@austinlparker would it be okay to put the @open-telemetry/cpp-maintainers as the "CODEOWNER" of this folder https://github.com/open-telemetry/opentelemetry.io/tree/main/content/en/docs/cpp? In this way we don't have to duplicate the contents in this repo.

I can update the [OTel C++ release process](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/RELEASING.md) and put something like "for each release, update the opentelemetry.io docs".

File: website_docs/_index.md
Comment (ThomsonTan) [reply website_docs/_index.md:21]: And I think Metrics should be "Not Yet Implemented".

**Comment by reyang** [comment]
I've submitted a PR https://github.com/open-telemetry/opentelemetry.io/pull/505. Once it is merged, I'll update the release process doc https://github.com/open-telemetry/opentelemetry-cpp/blob/main/RELEASING.md then we can close this PR.

**Review by reyang — CHANGES_REQUESTED** [review]
I wish to explore if we could avoid duplicating docs. Mirroring the docs will create problem since we don't have the same CI rule between the OTel C++ and opentelemetry.io repo.

**Comment by austinlparker** [comment]
I'm trying to avoid mirroring by making this be the authoritative source - my goal is to have a github action that will push these changes to the website each release.

**Comment by austinlparker** [comment]
Unfortunately this is one of those things where everyone wants to do this slightly differently :|

**Comment by reyang** [comment]
> I'm trying to avoid mirroring by making this be the authoritative source - my goal is to have a github action that will push these changes to the website each release.

This would be great!

**Comment by reyang** [comment]
@austinlparker I'm closing this PR as a result of #639.
If there is a way to automatically publish doc from this repo to opentelemetry.io (so we can keep all C++ related docs here), please let us know and we're happy to do it.

**Pull request closed without merging**
